test/scenarios.py

```diff
@@ -1231,9 +1231,124 @@
+                            {
+                                event.File(
+                                    FILES["testfile-big"].id, "testfile-big", 10485760
+                                ),
+                            },
+                        )
+                    ),
+                    action.CancelTransferRequest(0),
+                    action.Wait(event.FinishTransferCanceled(0, False)),
+                    action.NoEvent(),
+                    action.Stop(),
+                ]
+            ),
+            "stimpy": ActionList(
+                [
+                    action.Start("172.20.0.15"),
+                    action.Wait(
+                        event.Receive(
+                            0,
+                            "172.20.0.5",
+                            {
+                                event.File(
+                                    FILES["testfile-big"].id, "testfile-big", 10485760
+                                ),
+                            },
+                        )
+                    ),
+                    action.Wait(event.FinishTransferCanceled(0, True)),
+                    action.Download(
+                        0,
+                        FILES["testfile-big"].id,
+                        "/tmp/received",
+                    ),
+                    action.Wait(
+                        event.FinishFileFailed(
+                            0, FILES["testfile-big"].id, Error.BAD_TRANSFER
+                        )
+                    ),
+                    action.NoEvent(),
+                    action.Stop(),
+                ]
+            ),
+        },
+    ),
+    Scenario(
+        "scenario4-12",
+        "Send a request with one file to a peer that's offline. Wait for Queued event and cancel the transfer from the sender side. Expect no events on the receiver once it comes online",
+        {
+            "ren": ActionList(
+                [
+                    action.Start("172.20.0.5"),
+                    action.NewTransfer("172.20.0.15", ["/tmp/testfile-big"]),
+                    action.Wait(
+                        event.Queued(
+                            0,
+                            {
+                                event.File(
+                                    FILES["testfile-big"].id, "testfile-big", 10485760
+                                ),
+                            },
+                        )
+                    ),
+                    action.CancelTransferRequest(0),
+                    action.Wait(event.FinishTransferCanceled(0, False)),
+                    action.NoEvent(),
+                    action.Stop(),
+                ]
+            ),
+            "stimpy": ActionList(
+                [
+                    action.Sleep(8),
+                    action.Start("172.20.0.15"),
+                    action.NoEvent(),
+                    action.Stop(),
+                ]
+            ),
+        },
+    ),
```

Comment:
@matszczygiel This test fails even though I would expect it to pass, the Queued event is not sent while there's no connection to the peer.

Queued should mean that the file path was ok, files were traversed and the transfer is OK, in this case, it's never emitted

Reply:
The basic check is done in the `new_transfer()` call. `Queued` means that we have the connection working

Reply:
I like thinking about Queued so it would mean "the paths and information you provided are OK, your provided IP is also in the right format, here are the files I aggregated". Then use the Started event to indicate the action going on. Combined with FILE-393 the API user would always know which transfer is going to be retried. What is your take on this?

Reply:
The IP, files, and paths are checked in the `new_transfer()` call so if there is a problem the user will get the error from the function call. So there is really no way to get success from `new_transfer()` and not receive `Queued` at the same time, so emitting it would be redundant. However, I do agree that the app might want to know the details about the transfer immediately since the `new_transfer()` does not report this kind of information.
The second question is: do we want to inform the app that we have a connection active or not?

In general, I think you might be right about emitting `Queued` always, but I'm not sure if leaving apps blind about the connection state is a good idea - maybe we can introduce another event?

Reply:
If the user adds a directory and sends to an offline peer he will be oblivious to the files aggregated until the user appears online. He could query that data from the storage though so it's not that big of a deal.

However as we have `Started` events, should we even introduce the third event in between? 
Queued - all good, here what's going to be done
Started - here's what's happening

I think it even works out nicely in automated retries - we would emit Queued events for all the transfers we would retry